### ISSUE TITLE
[BUG] Delivery URL returned response code: 400

### ISSUE THREAD

**Issue by adjaruli** [dvdl16/woocommerce_fusion#130]
### Current Behavior

I have tried to set up the webhook correctly according to the documentation. If I want to save it, this error occurs in woocommerce (as can be seen in the picture) and there is no synchronization with ERPnext. Only the calls from ERPnext to woocommerce work and a manual transfer of the orders is then possible. I use frappe.cloud. I could not recognize anything from the logs regarding this error. I hope someone can help me, as I assume there must be an little error in my configuration.

![2024-12-03 11-34-33](https://github.com/user-attachments/assets/4d54fbb6-cb73-47df-83e0-57acad28b323)

### Steps To Reproduce

-

### Expected Behavior

-

### Environment and Versions

```markdown
- Frappe Version: v15.48.1
- ERPNext Version: v15.44.0
- Custom App Version: v1.7.4
```

### Operating System

Other

**Comment by LavaLines** [comment]
+1, facing the same error. Delivery URL is reachable, secret directly copied from WooCommerce-Server setting in ERPNext.

**Comment by dvdl16** [comment]
> there is no synchronization with ERPnext
Can you elaborate on what you mean by that?

@LavaLines do you have any logs? Have you checked the bullet points under the *Troubleshooting* section here: https://woocommerce-fusion-docs.finfoot.tech/features/sales-order.html#troubleshooting

**Comment by LavaLines** [comment]
> > there is no synchronization with ERPnext
> > Can you elaborate on what you mean by that?
> 
> [@LavaLines](https://github.com/LavaLines) do you have any logs? Have you checked the bullet points under the _Troubleshooting_ section here: https://woocommerce-fusion-docs.finfoot.tech/features/sales-order.html#troubleshooting

Thanks for getting back, unfortunately I don't now which logs would provide help finding the error. Google said there could be a webhook log under WooCommerce - Status - Logs, but there isn't. 
I did look through you troubleshooting, but hence the API is running just fine, all checks inside ERPNext look okay.
When checking doctype WooCommerce Order List I did notice, that orders from WooCommerce got picked up by the API, but they are not syncing to Sales Orders automatically. When manually pushed via Actions - Sync this Order to ERPNext they will appear as Sales Orders.
Is this normal behavior without the webhook working or a separate error?

**Comment by dvdl16** [comment]
There are 3 scenarios where a Sales Order will be created from a WooCommerce Order:
- Manually through the UI (Actions - Sync this Order to ERPNext), like you mentioned
- Every hour, the "sync_woocommerce_orders_modified_since" background task will run. Try resetting the "Last Sync Date" on "WooCommerce Integration Settings" to an earlier date, to see if you orders sync'ed
- If hourly is not frequent enough, you can configure the webhook

**Comment by LavaLines** [comment]
Wait - after fixing #159 the webhook seems to be working, too. Despite still showing response code 400 when saving the webhook settings, my test order got into sales orders inside ERPNext right after the order was made.

**Comment by dvdl16** [comment]
Yeah I don't know under what conditions WordPress/WooCommerce shows the `400`, so I'd say it's safe to ignore if you see orders being created in ERPNext

**Comment by jslocomotor** [comment]
Joining back here after becoming "ghost" 😄 
I still ran into this and did some debugging. The endpoint itself is reachable and the issue doesn't seem to be related to connectivity.

When creating the webhook, WooCommerce sends an initial ping containing webhook_id instead of an actual order payload. WooCommerce Fusion seems to process this request as a regular order and eventually tries to access order['id'], resulting in:

`
TypeError: byte indices must be integers or slices, not str
`

The traceback points to woocommerce_endpoint.py:

```python
f"{parse_domain_from_url(webhook_source_url)}{WC_RESOURCE_DELIMITER}{order['id']}"
```

A possible fix would be to detect the initial webhook_id ping and return a successful response before trying to process the payload as an order, e.g. by checking for webhook_id= in the request body.

I can still reproduce the issue with WooCommerce Fusion 1.17.3 / ERPNext 15.119.0 / Frappe 15.117.0.
